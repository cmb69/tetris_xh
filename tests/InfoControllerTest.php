<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Tetris_XH.
 *
 * Tetris_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Tetris_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tetris_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Tetris;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Tetris\Infra\HighscoreService;
use Tetris\Infra\SystemChecker;
use Tetris\Infra\View;

class InfoControllerTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $text = XH_includeVar("./languages/en.php", "plugin_tx")["tetris"];
        $systemChecker = $this->createStub(SystemChecker::class);
        $systemChecker->method("checkVersion")->willReturn(false);
        $systemChecker->method("checkExtension")->willReturn(false);
        $systemChecker->method("checkPlugin")->willReturn(false);
        $systemChecker->method("checkWritability")->willReturn(false);
        $highscoreService = $this->createStub(HighscoreService::class);
        $highscoreService->method("dataFolder")->willReturn("./content/");
        $view = new View("./views/", $text);
        $sut = new InfoController("./plugins/", $systemChecker, $highscoreService, $view);
        $response = $sut->defaultAction();
        Approvals::verifyHtml($response->output());
    }
}
