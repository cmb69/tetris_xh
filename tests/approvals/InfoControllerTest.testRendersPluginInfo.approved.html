<!-- tetris plugin info -->
<h1>Tetris</h1>
<img src="./plugins/tetris.png" class="tetris_logo" alt="Tetris cube">
<p>Version: 2.0-dev</p>
<p>Tetris_XH is powered by <a
  href="http://www.cmsimple-xh.org/wiki/doku.php/extend:jquery4cmsimple"
  target="_blank">jQuery4CMSimple</a> and <a
  href="http://fmarcia.info/jquery/tetris/tetris.html" target="_blank">
  Tetris with jQuery</a>.
</p>
<p>Copyright &copy; 2011-2017 Christoph M. Becker</p>
<p class="tetris_license">
  Tetris_XH is free software: you can redistribute it and/or modify it under
  the terms of the GNU General Public License as published by the Free
  Software Foundation, either version 3 of the License, or (at your option)
  any later version.
</p>
<p class="tetris_license">
  Tetris_XH is distributed in the hope that it will be useful, but <em>without
  any warranty</em>; without even the implied warranty of
  <em>merchantability</em> or <em>fitness for a particular purpose</em>. See
  the GNU General Public License for more details.
</p>
<p class="tetris_license">
  You should have received a copy of the GNU General Public License along with
  Tetris_XH. If not, see http://www.gnu.org/licenses/.
</p>
<div class="tetris_syscheck">
  <h2>System check</h2>
  <p class="xh_fail">Checking that PHP version ≥ 7.1.0 … failure</p>
  <p class="xh_fail">Checking that the PHP extension 'json' is loaded … failure</p>
  <p class="xh_fail">Checking that CMSimple_XH version ≥ 1.7.0 … failure</p>
  <p class="xh_fail">Checking that the CMSimple_XH plugin 'jquery' is installed … failure</p>
  <p class="xh_warning">Checking that './plugins/css/' is writable … warning</p>
  <p class="xh_warning">Checking that './plugins/config/' is writable … warning</p>
  <p class="xh_warning">Checking that './plugins/languages/' is writable … warning</p>
  <p class="xh_warning">Checking that './content/' is writable … warning</p>
</div>
